src/ServiceControl.Transports.ASBS/AzureQuery.cs

```diff
@@ -272,12 +265,37 @@ public override async IAsyncEnumerable<IBrokerQueue> GetQueueNames(
         var namespaces =
             subscription.GetServiceBusNamespacesAsync(cancellationToken);
 
-        await foreach (var serviceBusNamespaceResource in namespaces.WithCancellation(
-                           cancellationToken))
+        await foreach (var serviceBusNamespaceResource in namespaces.WithCancellation(cancellationToken))
         {
             if (validNamespaces.Contains(serviceBusNamespaceResource.Data.Name))
             {
                 resourceId = serviceBusNamespaceResource.Id;
+
+                // Determine the region of the namespace
+                var regionName = serviceBusNamespaceResource.Data.Location.Name;
+
+                // Build the regional Azure Monitor Metrics endpoint from the audience
+                var newEndpoint = BuildMetricsEndpointFromAudience(metricsQueryAudience, regionName);
+
+                // Create or refresh the MetricsClient if it's missing or points to a different region
+                if (client is null || metricsEndpoint?.ToString() != newEndpoint.ToString())
```

Comment:
I am unsure whether we need to handle the scenario where the Azure servicebus namespace has moved to a different region during runtime. Is there a bug associated with this change?
The move would have crashed ServiceControl, right?